### ISSUE TITLE
applyOverlap function query and subject confusion

### ISSUE THREAD

**Issue by Gregory94** [gacatag/IntEREst#18]
Hi!

I have some confusion about the `applyOverlap()` function.
I have two `SummarizedExperiment` objects:

1. `intRetObj` containing intron retention values based on a reference (UCSC) with collapsed exons;
2. `intSpanObj` containing intron spanning values based on a reference with uncollapsed exons.

In help of the function you are mentioning that "_The number of the rows and the dimensions of the counts of the result are equal to those of the subject. The function is applied on the query based on it's overlap to the subject_".
Maybe I'm misunderstanding the last line, but I would say that the `query` in this case should be the `intRetObj` and `subject` then be `intSpanObj` to "_map the reference with collpased exon to the reference with uncollapsed exon_" (as you mention in your vignette).
When I do this I run into issues when cbinding according to what you suggest in the vignette:
`cbind(query = intRetObj, subject = intRetSpanObj)` (where `intRetSpanObj` is the output of the `applyOverlap` function).
The problem appears to come forth from differences in chromosome names between `intRetObj` and `intRetSpanObj` (which on one hand sounds obvious because what is mentioned in the help text of the function: "_the number of the rows and the dimensions of the counts of the result are equal to those of the subject_"). When I switch the order of the query and subject, the cbinding works fine, but I'm not sure if that is the correct way of using that function.

Could you help clarifying how to correctly use the `applyOverlap` function in this case?

Thanks in advance!

Best,
Gregory

**Comment by gacatag** [comment]
Hi there!
Sorry fro the slight delay, I am on holiday and can't check emails often.
I think it is better if I describe what do we use the `applyOverlap` function for when performing differential intron retention analysis.

For differential intron retention analysis, the goal is to compare the changes to the number of reads that map to the introns relative to the changes to the number of reads that span the introns (e.g. when comparing two sets of sample). Furthermore, for the former (reads that map the introns), we would like to count the reads that map to the 'purely intronic regions' within the introns that do not overlap coordinates of exons of any isoforms. This is where `applyOverlap` can be used. The 'subject' is based on the 'normal' reference which includes the coordinates of the introns and exons. The 'query' is based on the same reference but with collapsed exonic regions. By collapse (or reduce) of exonic regions I mean that  all exonic coordinates associated with various isoforms of a gene are merged and the remaining coordinates are considered as 'purely intronic'. The reads that map to these purely intronic regions are counted using `interest` function. The read counts of all purely intronic regions that overlap an intron are summed using the `applyOverlap` function. This can be achieved by defining the object based on collapsed exonic regions (that also have the read count info) as the query, and the object based on normal reference (with uncollapsed exons !) as the subject. However, the chromsome names should match! I wonder why they are different ! Furthermore, the same resource (e.g. either RefSeq, ENSEMBL or etc.) should be used for building both references ! Do these info help solving the problem ?

Cheers,

Ali

**Comment by Gregory94** [comment]
Thank you for the explanation! Based on this I believe I use the `applyOverlap` function correctly then.
The query object is based on the collapsed exons reference which  I have build like this:
```
referencePrepare(sourceBuild="UCSC",
                               ucscGenome=referenceType,
                               collapseExons=TRUE,
                               fileFormat="gff3", annotateGeneIds=TRUE)
```
and the subject object is based on the uncollapsed exons reference:
```
referencePrepare(sourceBuild="UCSC",
                               ucscGenome=referenceType,
                               collapseExons=FALSE,
                               fileFormat="gff3", annotateGeneIds=TRUE)
```
The dimensions of both references are already wildly different (480540 vs 1401820 introns and exons, respectively). Do you know if this is normal?
Based on your explanation of what the `applyOverlap` function does I now performed the cbinding with the applyOverlap output object and the object based on the uncollapsed reference (that has been analyzed using the `IntSpan` method in the `interest` function). That does work, but previously I have performed the cbinding based on the object build with the collapsed reference (analyzed using the 'IntRet' methods in the `interest` function) based on what I understood from the vignette. That does not work because the dimensions do not match from the object with collapsed reference and the applyOverlap output object.

In short, do I understand correctly that the cbinding is to be performed with the applyOverlap output and the object from the uncollapsed reference?

Enjoy your holiday!

Best,
Gregory

**Comment by gacatag** [comment]
Hi Gregory! 
Yes, you understood correctly. I need to recheck the vignette. I have described the process in [Our recommended pipeline for differential intron retention analysis](https://bioconductor.org/packages/devel/bioc/vignettes/IntEREst/inst/doc/IntEREst.html#difanalysis). However, in the example, I did not use the `applyOverlap` and used `mdsChr22Obj` itself. This is probably due to the fact that genes that are included in the `mdsChr22Obj` do not overlap one-another, so there really isn't any difference between the collapsed exon and uncollapsed exon references ! If true , this would be an exception and should be clarified in the vignette. Thank you for raising the issue !

Cheers,

Ali

**Comment by Gregory94** [comment]
Thanks for the clarification!
